src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ThreadPool.Windows.cs

```diff
@@ -405,5 +411,35 @@ internal static unsafe void RequestWorkerThread()
             registeredWaitHandle.RestartWait();
             return registeredWaitHandle;
         }
+
+        private static unsafe void NativeOverlappedCallback(object? obj)
+        {
+            Debug.Assert(obj != null);
+            NativeOverlapped* overlapped = (NativeOverlapped*)(IntPtr)obj;
+            _IOCompletionCallback.PerformSingleIOCompletionCallback(overlapped, 0, 0);
+        }
+
+        [CLSCompliant(false)]
+        [SupportedOSPlatform("windows")]
+        public static unsafe bool UnsafeQueueNativeOverlapped(NativeOverlapped* overlapped)
+        {
+            // OS doesn't signal handle, so do it here (CoreCLR does this assignment in ThreadPoolNative::CorPostQueuedCompletionStatus)
+            overlapped->InternalLow = (IntPtr)0;
+            // Both types of callbacks are executed on the same thread pool
+            return UnsafeQueueUserWorkItem(NativeOverlappedCallback, (IntPtr)overlapped);
+        }
```

Comment:
You can avoid the `IntPtr`'s boxing allocation:

```suggestion
        private static unsafe void NativeOverlappedCallback(IntPtr ptr)
        {
            NativeOverlapped* overlapped = (NativeOverlapped*)ptr;
            _IOCompletionCallback.PerformSingleIOCompletionCallback(overlapped, 0, 0);
        }

        [CLSCompliant(false)]
        [SupportedOSPlatform("windows")]
        public static unsafe bool UnsafeQueueNativeOverlapped(NativeOverlapped* overlapped)
        {
            // OS doesn't signal handle, so do it here (CoreCLR does this assignment in ThreadPoolNative::CorPostQueuedCompletionStatus)
            overlapped->InternalLow = (IntPtr)0;
            // Both types of callbacks are executed on the same thread pool
            return UnsafeQueueUserWorkItem<IntPtr>(NativeOverlappedCallback, (IntPtr)overlapped, false);
        }
```

Reply:
Fixed in latest commit